### PR TITLE
linux-aarch64 no longer allows for 16K page size

### LINKED ISSUE #26: linux-aarch64 no longer allows for 16K page size
From https://github.com/microsoft/vscode/issues/179907

Since we started using musl to build linux-aarch64, it uses jemalloc, which doesn't support page sizes above 4K.

### PR DESCRIPTION
**Pull request by andreamah**
Fixes #26

### CODE REVIEW

File: build/build.sh [hunk 1]
```diff
@@ -10,23 +10,27 @@ set -ex
...
         "$CARGO" build --release --target=$TARGET
     else
         # Technically, MUSL builds will force PCRE2 to get statically compiled,
         # but we also want PCRE2 statically build for macOS binaries.
-        PCRE2_SYS_STATIC=1 "$CARGO" build --target "$TARGET" --release --features 'pcre2'
+        if is_linux && is_musl; then
+            JEMALLOC_SYS_WITH_LG_PAGE=14 PCRE2_SYS_STATIC=1 "$CARGO" build --target "$TARGET" --release --features 'pcre2'
+        else 
+            PCRE2_SYS_STATIC=1 "$CARGO" build --target "$TARGET" --release --features 'pcre2'
+        fi
```
Comment (deepak1556) [inline build/build.sh:33]: x86_64 does not need the page size change since it does not support.

File: build/build.sh [hunk 2]
```diff
@@ -10,15 +10,15 @@ set -ex
...
-    if is_linux && is_musl; then
-        # jemalloc doesn't allow 16K page sizes for linux musl 
-        # https://github.com/microsoft/ripgrep-prebuilt/issues/26
-        JEMALLOC_SYS_WITH_LG_PAGE=14
-    fi
-
     # Test a normal debug build.
     if is_arm || is_aarch64 || is_ppc64le; then
-        "$CARGO" build --target "$TARGET" --release --features 'pcre2'
+        if is_linux && is_musl; then
```
Comment (deepak1556) [inline build/build.sh:15]:
```suggestion
        if is_linux && is_aarch64 && is_musl; then
```

Comment (andreamah) [reply build/build.sh:15]: Do we need to run this flag for i686 linux? seems we have a musl build for that also.

Comment (deepak1556) [reply build/build.sh:15]: Nope it does not support 16KiB page size

File: build/build.sh [hunk 3]
```diff
@@ -10,15 +10,15 @@ set -ex
...
-    if is_linux && is_musl; then
-        # jemalloc doesn't allow 16K page sizes for linux musl 
-        # https://github.com/microsoft/ripgrep-prebuilt/issues/26
-        JEMALLOC_SYS_WITH_LG_PAGE=14
-    fi
-
     # Test a normal debug build.
     if is_arm || is_aarch64 || is_ppc64le; then
-        "$CARGO" build --target "$TARGET" --release --features 'pcre2'
+        if is_linux && is_aarch64 && is_musl; then
```
Comment (deepak1556) [inline build/build.sh:15]:
```suggestion
        if is_linux && is_aarch64_musl; then
```

Comment (deepak1556) [reply build/build.sh:15]: sorry this should be changed, refs https://github.com/microsoft/ripgrep-prebuilt/blob/aa3a29dd462cd264dd4d5a302b8309660fc3df97/build/utils.sh#L108C1-L113 , `is_aarch64` is only returned for the gnu build.

File: build/build.sh [hunk 4]
```diff
@@ -10,19 +10,17 @@ set -ex
...
-    fi
-
     # Test a normal debug build.
     if is_arm || is_aarch64 || is_ppc64le; then
         "$CARGO" build --target "$TARGET" --release --features 'pcre2'
     # pcre2 is not supported on s390x
     # https://github.com/zherczeg/sljit/issues/89
     elif is_s390x; then
         "$CARGO" build --release --target=$TARGET
+    elif is_aarch64_musl; then
```
Comment (andreamah) [inline build/build.sh:20]: would this never fall into the first condition?

Comment (deepak1556) [reply build/build.sh:20]: the targets are dependent on the architecture variable and you can see that aarch64 , aarch64-musl are distinct https://github.com/microsoft/ripgrep-prebuilt/blob/aa3a29dd462cd264dd4d5a302b8309660fc3df97/build/utils.sh#L22-L46

**Review by deepak1556 — APPROVED** [review]

File: build/build.sh (hunk 4 above)
Comment (deepak1556) [reply build/build.sh:20]: You can check that the env is applied correctly at https://dev.azure.com/vscode/ripgrep-prebuilt/_build/results?buildId=95112&view=logs&j=29bcf809-c0df-5a9c-77e9-04c1800335da&t=9228d288-de8e-50d8-cf3b-6c357507fd2d

**Review by rzhao271 — APPROVED** [review]

**Pull request merged**